### ISSUE TITLE
Warnings

### ISSUE THREAD

**Issue by dpo** [JuliaSmoothOptimizers/CUTEst.jl#80]
I'm getting the warnings
```
WARNING: Method definition cchprods!(Int64, Int64, Bool, Array{Float64, 1}, Array{Float64, 1}, Int64, Array{Float64, 1}, Array{Int64, 1}, Array{Int64, 1}) in module CUTEst at /Users/dpo/.julia/v0.5/CUTEst/src/specialized_interface.jl:5294 overwritten at /Users/dpo/.julia/v0.5/CUTEst/src/specialized_interface.jl:5304.
WARNING: Method definition cchprods!(CUTEst.CUTEstModel, Bool, Array{Float64, 1}, Array{Float64, 1}, Int64, Array{Float64, 1}, Array{Int64, 1}, Array{Int64, 1}) in module CUTEst at /Users/dpo/.julia/v0.5/CUTEst/src/specialized_interface.jl:5351 overwritten at /Users/dpo/.julia/v0.5/CUTEst/src/specialized_interface.jl:5363.
```

Also
```
WARNING: Method definition f(Any) in module Main at /Users/dpo/.julia/v0.5/CUTEst/test/hs32.jl:4 overwritten at /Users/dpo/.julia/v0.5/CUTEst/test/hs4.jl:4.
WARNING: Method definition g(Any) in module Main at /Users/dpo/.julia/v0.5/CUTEst/test/hs32.jl:5 overwritten at /Users/dpo/.julia/v0.5/CUTEst/test/hs4.jl:5.
WARNING: Method definition H(Any) in module Main at /Users/dpo/.julia/v0.5/CUTEst/test/hs32.jl:8 overwritten at /Users/dpo/.julia/v0.5/CUTEst/test/hs4.jl:6.
WARNING: Method definition #H(Array{Any, 1}, Main.#H, Any) in module Main overwritten.
```
We should probably give each problem function names other than just `f`, `g` and `H`.

**Comment by abelsiqueira** [comment]
I'm looking into the first problem.

`f` and friends are remnants from the days of yore. I'll remake the initial tests to use `NLPModels`.

These will be two PRs after 0.5.

**Comment by abelsiqueira** [comment]
Okay, the problem happens in the script because when
- in an inplace method,
- an output variable
- is an Array{Int, 1}
then I create two methods: one using Array{Int, 1} and one using Array{Cint, 1}.
However, for `cchprod`, the variable is `INOUT`, so created two versions using the same Array{Int, 1}.

A solution is removing the output requirement. This implies that inplace methods with any argument being array of int will have two methods: one for Array{Int, 1} in both input and output, and one for Array{Cint, 1} in both input and ouput.

Another solution would be removing this Cint version altogether.

**Comment by abelsiqueira** [comment]
Closed by #82 and #90
